tests/rptest/tests/archival_test.py

```diff
@@ -295,6 +295,93 @@ def test_all_partitions_leadership_transfer(self):
         self.kafka_tools.produce(self.topic, 5000, 1024)
         validate(self._cross_node_verify, self.logger, 90)
 
+    @cluster(num_nodes=3)
+    def test_timeboxed_uploads(self):
+        """This test checks segment upload time limit. The feature is enabled in the
+        configuration. The configuration defines maximum time interval between uploads.
+        If the option is set then redpanda will start uploading a segment partially if
+        configured amount of time passed since previous upload and the segment has some
+        new data.
+        The test sets the timeout value to 1s. Then it uploads data in batches with delays
+        between the batches. The segment size is set to 1GiB. We upload 10MiB total. So
+        normally, there won't be any data uploaded to Minio. But since the time limit for
+        a segment is set to 1s we will see a bunch of segments in the bucket. The offsets
+        of the segments won't align with the segment in the redpanda data directory. But
+        their respective offset ranges should align and the sizes should make sense.
+        """
+        self._extra_rp_conf.update(
```

Comment:
nit: If you don't want to stop and start the nodes in the test, you could use the `function_name` member within test_context to customize the configuration based on the test. For example, inside ArchivalTest constructor:
```
if test_context.function_name == "test_timeboxed_uploads":
    self._extra_rp_conf = dict(..stuff..goes..here...)
```
That way you don't have to stop and start the nodes yourself.

Reply:
This is very cool, thank you!